### PR TITLE
Added cloudbuild.yaml to push dns images.

### PR DESCRIPTION
**Pull request by prameshj**
Updated the image registry for pushing dns images.
Also updated the docs to reflect the new steps to push images.

@MrHohn @listx

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    - PULL_BASE_REF=$_PULL_BASE_REF
+    args:
+    - -c
+    - |
+      echo "Building/Pushing DNS containers"
+      make all-push
+substitutions:
+  # _GIT_TAG will be filled with a git-based tag for the image, of the form vYYYYMMDD-hash, and
+  # can be used as a substitution
+  _GIT_TAG: '12345'
```
Comment (MrHohn) [inline cloudbuild.yaml:19]: Do we need to modify this when releasing or will it be automatically filled?

File: cloudbuild.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# See https://cloud.google.com/cloud-build/docs/build-config
+
+# this must be specified in seconds. If omitted, defaults to 600s (10 mins)
+timeout: 3600s
+steps:
+  - name: 'gcr.io/k8s-testimages/gcb-docker-gcloud:v20190906-745fed4'
```
[the remark this answers is not in the document]
Comment (prameshj) [reply cloudbuild.yaml:6]: Good point, I see all the cloudbuild yamls using this image. @listx Do we anticipate this changing frequently?

File: cloudbuild.yaml (hunk 1 above)
Comment (prameshj) [reply cloudbuild.yaml:19]: I think we don't need to populate this, when we call make all-push after the tag, it works fine. I need to trigger the whole flow once to be sure though. I was not able to run this by hand due to permission issues. So I have added the prow yaml in https://github.com/kubernetes/test-infra/pull/17123 which is the recommended way.

File: cloudbuild.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (listx) [reply cloudbuild.yaml:6]: I haven't had the need to update this image for [the promoter](https://github.com/kubernetes-sigs/k8s-container-image-promoter/blob/master/cloudbuild.yaml) a while. So, you probably won't have to update too often either. But I'm just one datapoint. @Katharine should we be updating this image more frequently than update-it-when-the-build-breaks?

**Pull request merged**